### ISSUE TITLE
Training infrastructure support (record/track loss, gradients, accuracy, etc)

### ISSUE THREAD

**Issue by nvukobratTT** [tenstorrent/tt-forge-fe#179]
### Summary
For easier e2e training, it's good to have different training measurements recorded, and an easy way to track different runs and corresponding results. 

One of the tools that can help out is [TensorBoard](https://www.tensorflow.org/tensorboard). 

### Details
At the moment, PyBuda/Buda can extract loss and gradient values during training. With these options, we should be able to easily port and record these details for usage within TensorBoard. 

Likewise, it'll be good to track evaluation accuracy during different numbers of training steps. However, this part is more on the point how to integrate inference run during training, performantly. 

Additionally, more details can be added/tracked, but these should be easy to add up once core information has been recorded. 

### Passing criteria
- Training loop sample with data recording works on CPU (if possible on PyBuda/BBE stack)
- The following details are recorded during the training: 
  - Loss
  - Gradients for each parameter (e.g. weights, bias, etc.)
  - Model eval accuracy (passing rate; inference)
